test/type_prop.cpp

```diff
@@ -14446,3 +14446,22 @@ TEST(type_prop, scale_shift)
     EXPECT_EQ(scale_shift_func->get_element_type(), element::f64);
     EXPECT_EQ(scale_shift_func->get_shape(), (Shape{3, 6}));
 }
+
+TEST(type_prop, add_bcast)
```

Comment:
In addition to this kind of stuff I think we'll want to add some tests for `bcast_merge` and the like to `test/partial_shape.cpp`.